src/components/utils/build-select-box-lists.js

```diff
@@ -0,0 +1,109 @@
+import { localeCompare, filterOsByArchitecture } from '_/helpers'
+
+/**
+ * Return a normalized and sorted list of data centers for use in a __SelectBox__ from
+ * the list of provided data centers.
+ */
+function createDataCenterList (dataCenters) {
+  const dataCenterList =
+    dataCenters
+      .toList()
+      .filter(datacenter => true) // TODO: canUserUseDataCenter permission check?
+      .map(dataCenter => ({
+        id: dataCenter.id,
+        value: dataCenter.name,
+      }))
+      .sort((a, b) => localeCompare(a.value, b.value))
+      .toJS()
+
+  return dataCenterList
+}
+
+/**
+ * Return a normalized and sorted list of clusters ready for use in a __SelectBox__ from
+ * the Map of provided clusters.
+ */
+function createClusterList (clusters, dataCenterId = null) {
+  const clusterList =
+    clusters
+      .toList()
+      .filter(cluster =>
+        cluster.get('canUserUseCluster') &&
+        (dataCenterId ? cluster.get('dataCenterId') === dataCenterId : true)
+      )
+      .map(cluster => ({
+        id: cluster.get('id'),
+        value: cluster.get('name'),
+        datacenter: cluster.get('dataCenterId'),
+      }))
+      .sort((a, b) => localeCompare(a.value, b.value))
+      .toJS()
+
+  return clusterList
+}
+
+/**
+ * Return a normalized list of iso files from the set of provided storage domains,
+ * optionally filtering based on the provided data center.
+ */
+function createIsoList (storageDomains, dataCenterId = null) {
+  const list = []
+
+  storageDomains
+    .toList()
+    .filter(storageDomain =>
+      storageDomain.has('files') &&
+      storageDomain.get('canUserUseDomain') &&
+      (dataCenterId ? storageDomain.getIn([ 'statusPerDataCenter', dataCenterId ]) === 'active' : true)
```

Comment:
ternary operator could get a little confusing, consider using multiline ternary operators?
```
(dataCenterId ?
    (storageDomain.getIn([ 'statusPerDataCenter', dataCenterId ]) === 'active') : 
    true)
```

Reply:
I think it is the same amount of confusing on a single line v multiple lines in this case.  But if I change the condition a touch to make the "value of null means to skip that part of the check", a multiline looks ok:
````js
    (dataCenterId === null
      ? true
      : storageDomain.getIn([ 'statusPerDataCenter', dataCenterId ]) === 'active')
````